cli/src/semgrep/dependency_aware_rule.py

```diff
@@ -109,9 +117,25 @@ def generate_unreachable_sca_findings(
     return non_reachable_matches, dep_rule_errors, targeted_lockfiles
 
 
+@lru_cache(maxsize=1000)
```

Comment:
```suggestion
@lru_cache(maxsize=100_000)
```

I usually do 100x expected cache size for a typical run, just to prevent crazy memory leaks